### ISSUE TITLE
K-Level or Stackelberg

### ISSUE THREAD

**Issue by feiyuxiaoThu** [PuYuuu/vehicle-interaction-decision-making#2]
According to the author of your listing references, in Crossroad it seems the Stackelberg Game modeling is more suitable, have you considered this opinion and compared the differences of two modeling methods.
You can refer to Nan Li's talk [Autonomy Talks - Nan Li: Game-theoretic Methods for Safe Autonomous Vehicles on Shared Roads](https://youtu.be/eWHFiEP2NcQ?si=2AOOykgKDdaPFFIK)

**Comment by PuYuuu** [comment]
Thanks for your suggestion. I will watch the talk you recommended and study the performance of Stackelberg Game on crossroad. Reply to you later.

**Comment by PuYuuu** [comment]
Yes, I noticed that the author concluded in talk that the Level-k model is suitable for _**Highway Scenarios**_, and the Stackelberg model is suitable for **_Intersection Scenarios_**.
Through the experiment, I observed that these two models are not opposite. **The level-0 vehicle will proceed as the Leader, while the Level-1 vehicle will yield as the Follower because of the extra consideration.** The authors also mention the two models can be further combined. I think the key is the Level-K model should not simply treat other agents as level-(k-1), but model each agent behavior individually. However, it seems the role of the level-k model is only to define several decision strategies.
What do you think about that？ @feiyuxiaoThu 
![屏幕截图 2024-05-26 140645](https://github.com/PuYuuu/vehicle-interaction-decision-making/assets/35495891/73a9723c-27ca-4f0c-bfbc-7a08db3922b4)

**Comment by feiyuxiaoThu** [comment]
Thanks for your detailed reply. Based on your discussion on the connection between k-level and lead-follower models, I think they are both formulation methods of the complex interective scenarios. Level-k mainly focuses on the multi aspects, while the leader-follower uses many two-agent interection models to simplyfy the problem.
It seems the level-k is more popular both in academic and in industry, and I am curious about the reason.

**Comment by ZGY2000224** [comment]
Dear author, i would like to ask some questions and hope you could reply.
1. Most of articles about intersection conflict based on game theory choose "fixed paths", what do you think of this method.
2. Is there any idea of doing "path and speed" game(in other words, not only doing the speed planning , but also joining the path planning at the same time) in the future.

**Comment by PuYuuu** [comment]
Sorry I'm still learning too, but I'd like to try to answer your questions.
1.  I don’t understand what you mean by “_fixed path_” here. Can you list the references?
2.  In fact, the trajectory obtained based on the action sampling method contains “path and speed” information. I guess what you want to say is whether to consider doing higher-quality path planning and speed planning like self-vehicle planning (such as Apollo) in decision-making interaction. In my opinion, the answer no. Because it is too "heavy" for prediction and decision-making.
Hope these can be helpful to you. Thanks.

**Comment by ZGY2000224** [comment]
First，thanks for your reply!
1. For the "fixed path", it means that most of the existing methods do not analyze potential collision conflicts between vehicle trajectories in the conflict region from a spatial–temporal perspective in microscopic level, but achieve collision avoidance and multivehicle coordination by longitudinal speed planning based on a macroscopic level with a simplified point mass model and predefined straight, left or right turn paths. 
(Ref: Collision-Free Cooperative Motion Planning and Decision-Making for Connected and Automated Vehicles at Unsignalized Intersections
Intention Prediction and Mixed Strategy Nash Equilibrium-Based Decision-Making Framework for Autonomous Driving in Uncontrolled Intersection)
 In the relevant reference about levek-k game, the path strategy is considered, so, on a certain level, this is not the "fixed path". 
2. Actually, many of the study focus on the decisoin-making interaction, so they choose the "fixed path". But in my opinion, the higher-quality path planning should be considered and there is also truth to what you say.
If you don't mind, i recommend some paper to you and hope to learn together
1. Cooperative Driving of Connected Autonomous Vehicles in Heterogeneous Mixed Traffic: A Game Theoretic Approach
2. Game Theoretic Application to Intersection
Management: A Literature Review
3.Decision Making for Connected Automated Vehicles at Urban Intersections Considering Social and Individual Benefits
Finally, thanks for your help!

**Comment by ZGY2000224** [comment]
First，thanks for your reply!

For the "fixed path", it means that most of the existing methods do not analyze potential collision conflicts between vehicle trajectories in the conflict region from a spatial–temporal perspective in microscopic level, but achieve collision avoidance and multivehicle coordination by longitudinal speed planning based on a macroscopic level with a simplified point mass model and predefined straight, left or right turn paths.
(Ref: Collision-Free Cooperative Motion Planning and Decision-Making for Connected and Automated Vehicles at Unsignalized Intersections
Intention Prediction and Mixed Strategy Nash Equilibrium-Based Decision-Making Framework for Autonomous Driving in Uncontrolled Intersection)
In the relevant reference about levek-k game, the path strategy is considered, so, on a certain level, this is not the "fixed path".
Actually, many of the study focus on the decisoin-making interaction, so they choose the "fixed path". But in my opinion, the higher-quality path planning should be considered and there is also truth to what you say.
If you don't mind, i recommend some paper to you and hope to learn together
a. Cooperative Driving of Connected Autonomous Vehicles in Heterogeneous Mixed Traffic: A Game Theoretic Approach
b. Game Theoretic Application to Intersection Management: A Literature Review
c.Decision Making for Connected Automated Vehicles at Urban Intersections Considering Social and Individual Benefits
Finally, thanks for your help!

**Comment by PuYuuu** [comment]
Copy that. I will take some time to read the papers you recommended and reply to you later. 👀

**Comment by ZGY2000224** [comment]
By the way, do you have any idea for developing a version of MATLAB, I'm really looking forward to it

**Comment by PuYuuu** [comment]
Sorry, there is no plan for this. In my opinion, python and matlab are very close programming languages and you can do it yourself by referring to python. If you have any questions, feel free to contact me. My email: yuu.pu@foxmail.com

**Comment by ZGY2000224** [comment]
I get it. Thank you very much！
